lib/ansible/modules/cloud/amazon/elb_application_lb.py

```diff
@@ -719,14 +719,20 @@ def create_or_update_elb_listeners(connection, module, elb):
     for listener_to_add in listeners_to_add:
         try:
             listener_to_add['LoadBalancerArn'] = elb['LoadBalancerArn']
-            connection.create_listener(**listener_to_add)
+            # Rules is not a valid parameter for create_listener
```

Comment:
Looks good! On line [744](https://github.com/ansible/ansible/pull/27333/files#diff-a1bcebad578f0e6bfd841049d327a587L744) can you please also fix that to be deepcopy(listeners) (so the invocation gets shown correctly)? 
And lines [691](https://github.com/ansible/ansible/pull/27333/files#diff-a1bcebad578f0e6bfd841049d327a587L691) and 692 need to be set equal for new_rule, not current_rule. After that is appears rules are being updated correctly for me. Thanks for working on this.